### ISSUE TITLE
Edit source of events

### ISSUE THREAD

**Issue by sebastienkb** [jumartin/Calendar#31]
Hi,

Currently the demo app loads the calendar from the device itself, and I did not notice a method to change the data source on controller level. Is there a way to set our own events, for example by parsing a custom JSON array to an NSArray? (assuming the items comply to a specific structure)

Thanks for all this work.

**Comment by jumartin** [comment]
You have to implement in your view controller the methods of the `MGCDayPlannerViewDataSource` protocol. Required ones are : 
```
-dayPlannerView:numberOfEventsOfType:atDate:
-dayPlannerView:viewForEventOfType:atIndex:date:
-dayPlannerView:dateRangeForEventOfType:atIndex:date
```

**Comment by sebastienkb** [comment]
Thanks for the info.

I needed to figure out the best structure of data for easiest access but I think I got most of it.
Assuming I have an event object called SKEvent, I settled with a `NSDictionary<NSString*, NSArray<SKEvent *>*>` as main data source.
The keys are `yyyyMMdd` formatted strings from each day sate and the value is a time-sorted array of `SKEvent` corresponding to that date. Then I would go like this:

1.
`-dayPlannerView:numberOfEventsOfType:atDate:` 
uses `count` of the array of the corresponding date.

2.
`-dayPlannerView:viewForEventOfType:atIndex:date:` 
searches in the dictionary for a `SKEvent *event` thanks to `objectAtIndex:index` on the array of the corresponding date, then returns a new `MGCStandardEventView` with its title filled with data in my `SKEvent`.

3.
`-dayPlannerView:dateRangeForEventOfType:atIndex:date:` 
searches in the dictionary for a `SKEvent *event` thanks to `objectAtIndex:index` on the array of the corresponding date, and then returns `[MGCDateRange dateRangeWithStart:event.beginDate end:event.endDate];`

I think it's worth giving a quick example of a custom data source in your demo app, like suggested in #20.
